src/device.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2015-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::path::Path;
+use std::ffi::CString;
+
+#[cfg(feature = "use_glib")]
+use glib::translate::*;
+use ffi;
+use ::enums::{
+    Content,
+    DeviceType,
+    ScriptMode,
+    Status,
+};
+use ::surface::Surface;
+
+#[derive(Debug)]
+pub struct Device(*mut ffi::cairo_device_t, bool);
+
+impl Device {
+    pub unsafe fn from_raw_none(ptr: *mut ffi::cairo_device_t) -> Device {
+        assert!(!ptr.is_null());
+        ffi::cairo_device_reference(ptr);
+        Device(ptr, false)
+    }
+
+    pub unsafe fn from_raw_borrow(ptr: *mut ffi::cairo_device_t) -> Device {
+        assert!(!ptr.is_null());
+        Device(ptr, true)
+    }
+
+    pub unsafe fn from_raw_full(ptr: *mut ffi::cairo_device_t) -> Device {
+        assert!(!ptr.is_null());
+        Device(ptr, false)
+    }
+
+    pub fn to_raw_none(&self) -> *mut ffi::cairo_device_t {
+        self.0
+    }
+
+    pub fn create<P: AsRef<Path>>(filename: P) -> Option<Device> {
+        unsafe {
+            let filename = filename.as_ref().to_string_lossy().into_owned();
+            let filename = CString::new(filename).unwrap();
+            let p = ffi::cairo_script_create(filename.as_ptr());
+            if p.is_null() {
+                None
+            } else {
+                Some(Self::from_raw_full(p))
+            }
+        }
+    }
+
+    pub fn from_recording_surface(&self, surface: &Surface) -> Status {
+        unsafe {
+            Status::from(ffi::cairo_script_from_recording_surface(self.to_raw_none(),
```

Comment:
You need to ensure that this is actually a recording surface, and it might be better placed in the `RecordingSurface` type or not?

Reply:
Recording surface would look like the other specific surface types we have, type-wise. I guess.

Reply:
There should be a type for the recording surface

Reply:
How can we get a recording surface actually?

Reply:
`cairo_recording_surface_create()`